### PR TITLE
Add indicator for objed state

### PR DESCRIPTION
**Pull request by flatwhatson**
Adds support for https://github.com/clemera/objed

### CODE REVIEW

File: doom-modeline-segments.el [hunk 1]
```diff
@@ -1430,6 +1432,32 @@ mouse-1: Display Line and Column Mode Menu"
...
+
+(defun doom-modeline-update-objed (_ &optional reset)
+  "Update `objed' status, inactive when RESET is true."
+  (setq doom-modeline--objed-active (not reset)))
+
+(setq objed-modeline-setup-func #'doom-modeline-update-objed)
+
+(doom-modeline-def-segment objed-state ()
+  "The current objed state."
+  (when (and doom-modeline--objed-active
```
Comment (seagle0128) [inline doom-modeline-segments.el]: Is there any check here? Like `objed-mode` or `objed-local-mode`? I am not sure if `doom-modeline--objed-active` is safe enough.

File: doom-modeline.el [hunk 2]
```diff
@@ -86,15 +87,15 @@
 
 (doom-modeline-def-modeline 'main
   '(bar workspace-name window-number evil-state god-state ryo-modal xah-fly-keys matches buffer-info remote-host buffer-position parrot selection-info)
-  '(misc-info persp-name lsp irc mu4e github debug fancy-battery minor-modes input-method buffer-encoding major-mode process vcs checker))
+  '(objed-state misc-info persp-name lsp irc mu4e github debug fancy-battery minor-modes input-method buffer-encoding major-mode process vcs checker))
```
Comment (seagle0128) [inline doom-modeline.el]: In my understanding, `objed-state` is kind of modal editing state, like `evil-state` or `god-state`, correct? If I am not wrong, it should be put to left side of mode-line, near `evil-state` and other modal editing indicators.

I've merged all editing state indicators into one `modal` segment in 2140ded3deec9c472081184a6e7ccdb03a5be5f3.

File: doom-modeline-segments.el (hunk 1 above)
Comment (flatwhatson) [reply doom-modeline-segments.el]: The module provides a modeline setup hook `objed-modeline-setup-func`, which is called to set the modeline on entry to the modal state, and called again with the reset flag when exiting.  Normally this runs a routine to modify the modeline directly, but we update a variable for the segment instead.

Ideally `objed` would provide a state indicator variable that could be used for this purpose, but it doesn't.  Other internal variables can't be relied upon as they're often left with the previous values even when leaving the modal state.  I don't mind submitting some improvements upstream, but the segment works reliably as-is in my testing for the last week.

File: doom-modeline.el (hunk 2 above)
Comment (flatwhatson) [reply doom-modeline.el]: Yes, it is a modal indicator, but it doesn't work well in any position to the left of the buffer name.  The indicator looks like `line(w)` or `paragraph(w)` and this changes frequently during editing, causing a big chunk of the modeline to jump around.  The position of `global-mode-string` works well for this indicator.

**Comment by flatwhatson** [comment]
Thanks for your review, I'll come back tomorrow to rebase & make any changes needed.

**Comment by seagle0128** [comment]
Okay. I merged it. Please maintain this segment since I am not using `objed`. Thanks!

**Comment by flatwhatson** [comment]
Yes will do.  Thank you!

**Pull request merged**